torch/_logging.py

```diff
@@ -0,0 +1,286 @@
+# Top level logging module for torch logging
+# Design doc: https://docs.google.com/document/d/1ZRfTWKa8eaPq1AxaiHrq4ASTPouzzlPiuquSBEJYwS8/edit#
+# Simple setup for onboarding (see above doc for more detail):
+# 1. register_log for all logs you'd like to have control of
+# 2. @loggable any classes you'd like to register as artifacts can be toggled as logged/not logged
+#    Only requirement here is that it has a __str__ method, and then instances of this class can be passed directly
+#    to log.debug(<instance here>)
+# 3. call init_logging([.. your log names .. ]) somewhere in your code
+# (before the user may attempt to set logs, but after you've executed 1. and 2.)
+
+import collections
+import itertools
+import logging
+import os
+import re
+from typing import DefaultDict, Dict, Set
+
+
+DEFAULT_LOG_LEVEL = logging.WARN
+DEFAULT_FORMATTER = logging.Formatter(
+    "[%(asctime)s] %(name)s: [%(levelname)s] %(message)s"
+)
+
+NAME_TO_LOG_NAME = {}
+NAME_TO_RECORD_TYPE = {}
+LOG_NAME_TO_REC_TYPES = collections.defaultdict(set)
+# the names of artifacts associated with each log
+# includes the log_name as well
+LOG_NAME_TO_NAMES = collections.defaultdict(set)
+# log names or artifact names can be part of the settings string
+# dynamo + inductor logs have verbosity settings, aot only has one level
+# names which support verbosity (prefix with a + lower or -)
+# NB: this is the setting name
+VERBOSE_NAMES = set()
+
+# Set by user-facing API
+name_to_level: Dict[str, int] = {}
+enabled_artifact_names: Set[str] = set()
+
+
+def is_initialized():
+    for log_name in LOG_NAME_TO_NAMES.keys():
```

Comment:
This seems weirdly expensive. Why not just have a global variable that tells you if you initialized or not.

Reply:
e.g. 

https://github.com/huggingface/transformers/blob/a9bd5df16a46356463f2712dd8f6c109fa83d6f9/src/transformers/utils/logging.py#L101

Reply:
Not needed anymore, since we initialize when torch is imported